src/vehicle.cpp

```diff
@@ -640,9 +631,13 @@ void vehicle::control_engines() {
 
     if( !dirty ) { return; }
 
+    auto eng = std::find_if( parts.begin(), parts.end(), []( const vehicle_part &e ) {
+        return e.is_engine() && e.enabled;
+    } );
+
     // if current velocity greater than new configuration safe speed
     // drop down cruise velocity.
-    int safe_vel = safe_velocity();
+    int safe_vel = safe_velocity( *eng ) * 2.237 * 100;
```

Comment:
Current `velocity` is still in arbitrary units so those conversions are needed. I'm going to drop them once `velocity` is in `m/s` however that will be a separate PR to avoid this one growing too large